### DISCUSSION TITLE
Lambda builder question

### DISCUSSION THREAD

**Discussion by bensteinberg** [dokku/dokku#5353] [Q&A] [answered]
Hi -- I'm trying to run a Python lambda in my Dokku instance. In my repo I have a `lambda.yml`:
```
---
build_image: mlupin/docker-lambda:python3.9-build
builder: python
run_image: mlupin/docker-lambda:python3.9
```
a `function.py`:
```
def handler(event, context):
    return {"Hello": "world!"}
 ```
 and `pyproject.toml` and `poetry.lock`. When I push to the `dokku` origin, I see the lambda builder working, correctly identifying the python set-up, installing dependencies via poetry, creating lambda.zip, building the runtime container based on `mlupin/docker-lambda:python3.9`, and deploying it. The nginx config looks right, and the logs report `Lambda API listening on port 5000...` -- but any POST to the application's URL gets a 405. Have I missed a step in setting this up? Is there a complete example of this procedure anywhere?
 
I can provide more detailed output from deployment or other commands if that'll help. Thanks!

**Comment by josegonzalez** [comment]
What is the exact curl request you are using to create the POST?

**Reply by bensteinberg** [reply]
Thanks. I'm running 

```curl -X POST "https://func.example.com/" -H "Content-type: application/json" --data '{"hi": "there"}'```

 except with my domain name.

**Reply by bensteinberg** [reply]
The response in the nginx log looks like
```
172.68.54.211 - - [10/Sep/2022:06:59:13 +0000] "POST / HTTP/1.1" 405 0 "-" "curl/7.85.0"
```

**Comment by josegonzalez** [comment, marked as answer]
For the lambda builder, the URL path you want to curl will be something like:

```
# where function.handler is your handler
# below works for this app: https://github.com/dokku/dokku/tree/master/tests/apps/lambda-python
/2015-03-31/functions/function.handler/invocations
```

This matches how the handlers respond for AWS Lambda, and the built images are compatible with AWS Lambda otherwise.

**Reply by bensteinberg** [reply]
Thanks so much, @josegonzalez! That does the trick. It looks like routes matching `/2015-03-31/functions/<anything>/invocations` will work.

**Reply by bensteinberg** [reply]
(And I think that the 405 Method Not Allowed response is from the attempt to serve a static 404 page in response to a POST.)
